superset-frontend/src/explore/components/DataTablesPane.jsx

```diff
@@ -0,0 +1,201 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import { styled, t } from '@superset-ui/core';
+import {
+  CopyToClipboardButton,
+  FilterInput,
+  RowCount,
+  useFilteredTableData,
+  useTableColumns,
+} from './DataTableControl';
+import { Collapse } from '../../common/components';
+import Tabs from '../../common/components/Tabs';
+import Loading from '../../components/Loading';
+import TableView, { EmptyWrapperType } from '../../components/TableView';
+import { getChartDataRequest } from '../../chart/chartAction';
+import getClientErrorObject from '../../utils/getClientErrorObject';
+
+const RESULT_TYPES = {
+  results: 'results',
+  samples: 'samples',
+};
+const NULLISH_RESULTS_STATE = {
+  [RESULT_TYPES.results]: null,
+  [RESULT_TYPES.samples]: null,
+};
+
+const TableControlsWrapper = styled.div`
+  display: flex;
+  align-items: center;
+
+  span {
+    flex-shrink: 0;
+  }
+`;
+
+const SouthPane = styled.div`
+  background-color: ${({ theme }) => theme.colors.grayscale.light5};
+  z-index: 1;
+`;
+
+const TabsWrapper = styled.div`
+  height: ${({ contentHeight }) => contentHeight}px;
+  overflow: hidden;
+
+  .table-condensed {
+    height: 100%;
+    overflow: auto;
+  }
+`;
+
+export const DataTablesPane = ({
+  queryFormData,
+  tableSectionHeight,
+  onCollapseChange,
+}) => {
+  const [data, setData] = useState(NULLISH_RESULTS_STATE);
+  const [isLoading, setIsLoading] = useState(NULLISH_RESULTS_STATE);
+  const [error, setError] = useState(NULLISH_RESULTS_STATE);
+  const [filterText, setFilterText] = useState('');
+  const [activeTabKey, setActiveTabKey] = useState(RESULT_TYPES.results);
+
+  useEffect(() => {
+    const getData = resultType => {
+      setIsLoading(prevIsLoading => ({ ...prevIsLoading, [resultType]: true }));
+      return getChartDataRequest({
+        formData: queryFormData,
+        resultFormat: 'json',
+        resultType,
+      })
+        .then(response => {
+          // Currently displaying of only first query is supported
```

Comment:
grammar nit: `Only displaying the first query is currently supported`

Reply:
Oops, that's what I get for copy pasting without checking 🙂 I'll fix it in DisplayQueryButton too.